errors.go

```diff
@@ -15,6 +15,8 @@ package notation
 
 // ErrorPushSignatureFailed is used when failed to push signature to the
 // target registry.
+//
+// Deprecated: Use PushSignatureFailedError instead.
 type ErrorPushSignatureFailed struct {
```

Comment:
Since `ErrorPushSignatureFailed` and `PushSignatureFailedError` are the exact same thing, can you do
```
type ErrorPushSignatureFailed = PushSignatureFailedError
```
The same comment applies to other types.

Reply:
> Since `ErrorPushSignatureFailed` and `PushSignatureFailedError` are the exact same thing, can you do
> 
> ```
> type ErrorPushSignatureFailed = PushSignatureFailedError
> ```
> 
> The same comment applies to other types.

@shizhMSFT updated, please let me know if it looks like you want

Reply:
LoL. I mean you can do something like
```go
// ErrorPushSignatureFailed is used when failed to push signature to the
// target registry.
//
// Deprecated: Use PushSignatureFailedError instead.
type ErrorPushSignatureFailed = PushSignatureFailedError

// PushSignatureFailedError is used when failed to push signature to the
// target registry.
type PushSignatureFailedError struct {
	Msg string
}

func (e PushSignatureFailedError) Error() string {
	if e.Msg != "" {
		return "failed to push signature to registry with error: " + e.Msg
	}
	return "failed to push signature to registry"
}
```
instead of putting it in the comment block.

Reply:
done